blocksync/reactor.go

```diff
@@ -66,22 +66,18 @@ type Reactor struct {
 
 // NewReactor returns new reactor instance.
 func NewReactor(state sm.State, blockExec *sm.BlockExecutor, store *store.BlockStore,
-	blockSync bool, metrics *Metrics,
+	blockSync bool, metrics *Metrics, offlineStateSyncHeight int64,
 ) *Reactor {
 
 	storeHeight := store.Height()
-	var err error
-	if storeHeight == 0 {
+	if offlineStateSyncHeight != 0 {
```

Comment:
```suggestion
	if storeHeight == 0 {
```

Reply:
I just added a fix for the code in `node.go` which should already take care of this check: https://github.com/cometbft/cometbft/pull/1057/files#r1285792914 

If storeHeight is not 0 `offlineStateSyncHeight` will be 0. But I don't mind repeating the check here.